D3D12Bundles/D3D12Bundles.cs

```diff
@@ -412,42 +415,41 @@ struct Vertex {
                     BlendState = BlendDescription.Opaque,  //Nothing seems to correspond to CD3DX12_BLEND_DESC(D3D12_DEFAULT)
                     //BlendState = BlendDescription.NonPremultiplied, //i.e. new BlendDescription(Blend.SourceAlpha, Blend.InverseSourceAlpha),
                     DepthStencilState = DepthStencilDescription.Default,
-                    DepthStencilFormat = Format.D32_Float,
+                    DepthStencilFormat = mPresenter.PresentationParameters.DepthStencilFormat,
                     SampleMask = uint.MaxValue, //This is the default value anyway...
                     PrimitiveTopologyType = PrimitiveTopologyType.Triangle,
-                    RenderTargetFormats = new[] { Format.R8G8B8A8_UNorm, },
+                    RenderTargetFormats = new[] { mPresenter.PresentationParameters.BackBufferFormat, },
                     SampleDescription = SampleDescription.Default,  //This is the default value anyway...
                 };
-                mPipelineState1 = mDevice.CreateGraphicsPipelineState(psoDesc);
-                mPipelineState1.Name = nameof(mPipelineState1);
-
-                // Modify the description to use an alternate pixel shader and create
-                // a second PSO.
-                psoDesc.PixelShader = pixelShader2;
-                mPipelineState2 = mDevice.CreateGraphicsPipelineState(psoDesc);
-                mPipelineState2.Name = nameof(mPipelineState2);
-            }
+                mPipelineState1 = new PipelineState(mGraphicsDevice.NativeDevice, mRootSignature, psoDesc1);
+                mPipelineState1.NativePipelineState.Name = nameof(mPipelineState1);
 
-            // Create the command list.
-            mCommandList = mDevice.CreateCommandList<ID3D12GraphicsCommandList>(0, CommandListType.Direct, mCommandAllocator, null);
-            mCommandList.Name = nameof(mCommandList);
-
-            // Create render target views (RTVs).
-            {
-                CpuDescriptorHandle rtvHandle = mRtvHeap.GetCPUDescriptorHandleForHeapStart();
-
-                for (int i = 0; i < FrameCount; i++) {
-                    mRenderTargets[i] = mSwapChain.GetBuffer<ID3D12Resource>(i);
-
-                    mDevice.CreateRenderTargetView(mRenderTargets[i], null, rtvHandle);
-                    rtvHandle = rtvHandle.Offset(1, mRtvDescriptorSize);
-                    mRenderTargets[i].Name = $"{nameof(mRenderTargets)}[{i}]";
-                }
+                // Duplicate the description but use an alternate pixel shader and create a second PSO.
+                var psoDesc2 = new GraphicsPipelineStateDescription {
+                    //InputLayout = new InputLayoutDescription(Vertex.InputElements),
+                    InputLayout = new InputLayoutDescription(FlatShadedVertex.InputElements),
+                    RootSignature = mRootSignature,
+                    VertexShader = vertexShader,
+                    PixelShader = pixelShader2,
```

Comment:
I'm tempted to subclass this and add a method to expose MemberwiseClone to avoid this duplication, or some other solution... but for now a quick & dirty copy-paste, which makes it much less obvious that the only intended change is this `L433`.

Before I could simply edit the original type and reuse it with another call to `CreateGraphicsPipelineState` because it was only used to create it.  But now with the higher abstraction, PipelineState holds a reference to this object, so I can't edit it without impacting the first pipeline state, otherwise its description won't match what was created.... hence the need for a new object now.